tools/certgen/README.md

```diff
@@ -0,0 +1,73 @@
+# CertGen
+
+This is a scripting tool for generating certificates on disk for end-to-end testing and local development. It is responsible for creating a _Certificate Authority_ (CA) and using it to issue certificates for client and server applications.
+
+To establish encrypted communication between the client and the server, it is necessary to use the TLS protocol and x.509 certificates, but some particularities of the certificates must be taken into account. Basically the functionality of this tool is to carefully generate certificates prepared for use in TLS connections between client and server, that is, the details of how the certificates are configured are the responsibility of this tool. What really matters to anyone using this tool is the _host name_ (or IP) of the server that the client application tries to connect to. It must match one of the _Subject Alternative Names_ in the certificate of the server (or, failing that, it must match the common name (CN) of the cert's Subject Distinguished Name (DN).
+
+With this tool, just inform the host names, IPs and CN of the certificate as parameters and it takes care to generate a ready-to-use certificate.
+
+It uses the [OpenSSL](https://www.openssl.org/) to generate certificates and cryptographic keys.
+
+## Usage
+
+This tool creates/uses the directories `./ca` and `./certs` to store the certificates and public/private keys of the CA and hosts respectively.
+
+###### Example 1
+
+Give the script execution permission (`chmod u+x certgen.sh`) and enter the command:
+
+~~~bash
+./certgen.sh \
+    --dns 'server.domain.com' \
```

Comment:
In the case of devices. You don't need to pass DNS and IP, maybe you can make it clear.

Reply:
What's happen if I don't pass DNS?

Reply:
Maybe We can have a example how create certificates for a device in dojot.

Reply:
For device certificates, you must use the x509-identity-mnmt component API. That's because VerneMQ only accepts certificates generated by the EJBCA.
The idea of this script is to be able to generate certificates for the services, since the x509-identity-mnmt component cannot do this.
This script creates its own CA and uses it to issue its certificates, so we can use these certificates in the services. We can store the certificates somewhere that can be reused, even when the environment is brought down and restarted.

Reply:
If the SAN extension is not present in the certificate (DNS and IP), the CN field of the SubjectDN is used to validate the domain.

Reply:
I will submit a PR with a script to generate certificates for devices (using the x509-identity-mnmt API). In this case, it would be a different tool, with its own README.

Reply:
The generate certs script in Locust already creates the certificates for devices.